R/S4.R

```diff
@@ -1,17 +1,44 @@
 #' Register an R7 class with S4
 #'
-#' If you want to use [method<-] to register an method for an S4 generic with
-#' an R7 class, you need to call `S4_register()` once.
+#' @description
+#' If you want to use and R7 class with S4 (e.g. to use [method<-] to register an
+#' method for an S4 generic with an R7 class) you need to call `S4_register()`
+#' once. This generates a full S4 class specification that:
+#'
+#' * Matches class name and inheritance hierarchy.
+#' * Uses [validate()] as the validity method.
+#' * Defines formal S4 slots to match R7 properties. The slot types are
+#'   matched to the R7 property types, with the exception of R7 unions,
+#'   which are unchecked (due to the challenges of converting R7 unions to
+#'   S4 unions).
+#'
+#' If `class` extends another R7 class or has a property restricted to an
+#' R7 class, you you must register those classes first.
```

Comment:
If you have a package where you are doing this (i.e. you have 2 R7 classes where one extends the other and you want them both to be S4 registered), this depends on the order that the files are sourced in, right? Since you call `S4_register()` globally in the files? I guess there isn't much we can do about that if that is the case, but it has potential to be confusing.

Reply:
Right, but for R7 inheritance to work you'd have to have sourced the parent before the child.